Fix/Add db:(schema/structure):load:with_data

craineum opened the pull request:
- Corrects `db_schema.rb` loading on `:load:with_data` tasks
- Add `db:structure:load:with_data` support
- Remove ignored file: `tasks/databases.rake` so search will be more accurate

Fixes #90

craineum commented:
I combined a few things that seemed related. Let me know if you want me to separate this out or if you want to me to address anything else.

EnomaDebby commented on lib/data_migrate/database_tasks.rb:
```diff
@@ -2,11 +2,12 @@
 
-    def self.data_schema_file
+    # This overrides ActiveRecord::Tasks::DatabaseTasks
+    def self.schema_file(_format = nil)
```
Curious why you changed this to `schema_file `since `data_schema_file` is more descriptive of the file

craineum replied:
The code was relying on the fact that it was overriding the method from `ActiveRecord::Tasks::DatabaseTasks`. Without this `schema_file` was not being called when `load_schema_current` is called from `tasks/databases.rake db:schema:load`

It is more descriptive but broke the original intent.

craineum commented on tasks/databases.rake:
```diff
@@ -229,7 +229,24 @@ namespace :db do
+    end
+  end
+
+  namespace :structure do
```
done

craineum commented:
Not sure what to do with this but in moving to 5.2 I found another bug. It will only occur on my branch since it fixes the `db:schema:load:with_data` but is unrelated to my changes or the original issue. The symptoms show when running `db:schema:load:with_data` (or in my case `db:structure:load:with_data`):
```
-- quote_table_name("data_migrations")
   -> 0.0001s
-- select_values("SELECT version FROM \"data_migrations\"")
   -> 0.0010s
-- quote_table_name("data_migrations")
   -> 0.0000s
-- execute("INSERT INTO \"data_migrations\" (version) VALUES ('20180511220109')")
   -> 0.0009s
rails aborted!
NoMethodError: undefined method `full_migrations_path' for DataMigrate::DataMigrator:Class
```
The root cause is that `lib/data_migrate/data_schema.rb:34` calls for `DataMigrate::DataMigrator.full_migrations_path` but it does not exist in `lib/data_migrate/data_migrator_five.rb` file.

I need this working so I am going to create a new branch and fix this issue separately. I can either then include that fix here or create a new bug that will be dependant on this fix being merged (or one like it).

craineum commented:
@ilyakatz The actual issue was that the missing methods are being stubbed when they don't actually exist. I added a rspec config that option that will break when that happens. I also fixed the issue(s).

ilyakatz commented on .travis.yml:
```diff
@@ -1,16 +1,12 @@
 rvm:
-  - 2.2.6
-  - 2.3.4
+  - 2.2.9
```
👍 nice, thanks for doing this

ilyakatz commented:
Deployed to rubygems
